vars/getTestResults.groovy

```diff
@@ -0,0 +1,141 @@
+        def testsPassed = testsTotal - testsFailed - testsSkipped
+        def failedTests = testResultAction.getFailedTests()
+
+        if (failedTests){
+            for (test in failedTests) {
+                def failDocument = ['build_number': buildNumber, 'pull_request': prString, 'build_description': remainingText, 'test_class': test.getParent().getName(), 'test_name': test.fullName, 'test_status': 'FAILED', 'build_result': buildResult, 'test_fail_count': testsFailed, 'test_skipped_count': testsSkipped, 'test_passed_count': testsPassed, 'build_duration': buildDuration, 'build_start_time': buildStartTime]
+                testResults.add(failDocument)
+            }
+        } else {
+            println("No test failed.")
+        }
+    }
+    return testResults
+}
+
+void indexFailedTestData() {
+
+    withCredentials([
+            string(credentialsId: 'jenkins-external-metrics-cluster-endpoint', variable: 'METRICS_HOST_URL')
+    ]) {
+
+        sh """
+        set +e
+        
+        MONTH_YEAR=\$(date +"%m-%Y")
+        INDEX_NAME="gradle-check-\$MONTH_YEAR"
+        INDEX_MAPPING='{
+                "mappings": {
+                    "properties": {
+                      "build_number": { "type": "integer" },
+                      "pull_request": { "type": "keyword" },
+                      "build_description": { "type": "text" },
+                      "test_class": { "type": "keyword" },
+                      "test_name": { "type": "keyword" },
+                      "test_status": { "type": "keyword" },
+                      "build_result": { "type": "keyword" },
+                      "test_fail_count": { "type": "integer" },
+                      "test_skipped_count": { "type": "integer" },
+                      "test_passed_count": { "type": "integer" },
+                      "build_duration": { "type": "float" },
+                      "build_start_time": { "type": "date" }
+                    }
+                }
+        }'
+        echo "INDEX NAME IS \$INDEX_NAME"
+        curl "${METRICS_HOST_URL}"
+        curl -I "${METRICS_HOST_URL}/\$INDEX_NAME" | grep -q "200 OK"
+        if [ \$? -eq 0 ]; then
+            echo "Index already exists. Indexing Results"
+        else
+            echo "Index does not exist. Creating..."
+            create_index_response=\$(curl -s -XPUT "${METRICS_HOST_URL}/\${INDEX_NAME}" -H 'Content-Type: application/json' -d "\${INDEX_MAPPING}")
+            if [[ \$create_index_response == *'"acknowledged":true'* ]]; then
+                echo "Index created successfully."
+            else
+                echo "Failed to create index. Error message: \$create_index_response"
+                exit 1
+            fi
+        fi
+        
+        if [ -s failed-test-records.json ]; then
+            echo "File Exists, indexing results."
+            curl -XPOST "${METRICS_HOST_URL}/\$INDEX_NAME/_bulk" -H "Content-Type: application/x-ndjson" --data-binary "@failed-test-records.json"
```

Comment:
How to get authenticated with the metrics host?

Reply:
This is still work in progress.